src/lib/ext-activitylog.js

```diff
@@ -194,6 +233,34 @@ class Controller {
     }
   }
 
+  async loadLogs(file) {
+    try {
+      const logStr = await load.loadLogAsText(file);
+      const logs = JSON.parse(logStr);
+
+      const storedFileName = await browser.runtime.sendMessage({
+        requestTo: 'ext-monitor',
+        requestType: 'setLoadedLogs',
+        detail: { fileName: file.name, logs },
+      });
+      const searchParams = `file=${storedFileName}`;
+
+      this.view.setError(null);
+
+      const tab = await browser.tabs.create({
+        url: getActivityLogPageURL(searchParams),
+      });
+
+      await browser.runtime.sendMessage({
+        requestTo: 'ext-monitor',
+        requestType: 'setLoadedLogsTabId',
+        detail: { tabId: tab.id, fileName: storedFileName },
+      });
```

Comment:
Instead on being the activitylog page to send the logs to the background page, then create a new tab and then send another message to the background page to associate the tab.id to the loaded file, this could actually be a responsibility of the background page (or the ext-monitor module actually), as part of what it does to handle a single 'openLoadedLogsTab`.

(Let me know if my request isn't clear enough as described above)

Reply:
So, in summary, I will be doing the following - 
- Send the loaded file from activitylog.html to the background.
- In the background, I will -
    - read the log file. 
    - open a new tab.
    - load the logs in that tab.

@rpl Am I right?

Reply:
yeah that's right (but I notice that you did already proceed to implement it, and so my reply did come late :-P)